src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java

```diff
@@ -73,6 +79,71 @@ public LoginResponseDto loginClient(String clientId, String clientSecret) throws
         .build();
   }
 
+  @Override
+  public void validateToken(String token) throws BadJOSEException, ParseException, JOSEException {
+    var decodedToken = Oauth2Util.getTokenDecoded(token, oauth2SecretKey);
+
+    final Date now = new Date();
+    final Date exp = decodedToken.getExpirationTime();
+
+    if (exp != null) {
```

Comment:
Don't do this by your own, have you tried using `DefaultJWTProcessor#setJWTClaimsSetVerifier`?

Docs: https://javadoc.io/doc/com.nimbusds/nimbus-jose-jwt/latest/com/nimbusds/jwt/proc/DefaultJWTProcessor.html#%3Cinit%3E()

```
A [default JWT claims verifier](https://javadoc.io/static/com.nimbusds/nimbus-jose-jwt/9.30.1/com/nimbusds/jwt/proc/DefaultJWTClaimsVerifier.html) is provided, to perform a minimal check of the claims after a successful JWS verification / JWE decryption. It checks the token expiration (exp) and not-before (nbf) timestamps if these are present. The default JWT claims verifier may be extended to perform additional checks, such as issuer and subject acceptance.
```
More docs:
https://javadoc.io/doc/com.nimbusds/nimbus-jose-jwt/latest/com/nimbusds/jwt/proc/DefaultJWTClaimsVerifier.html

Example:
https://connect2id.com/products/nimbus-jose-jwt/examples/validating-jwt-access-tokens

Reply:
Done.